panther_hardware_interfaces/src/roboteq_data_converters.cpp

```diff
@@ -122,6 +123,15 @@ panther_msgs::msg::FaultFlag FaultFlag::GetMessage() const
   return fault_flags_msg;
 }
 
+std::map<std::string, bool> FaultFlag::GetErrorMap() const
+{
+  std::map<std::string, bool> error_map;
+  for (std::size_t i = 0; i < flag_names_.size(); i++) {
+    error_map["fault_flag." + flag_names_[i]] = flags_.test(i);
+  }
+  return error_map;
+}
+
```

Comment:
Wonder if we could define this method in `FlagError` with optional parameter map_prefix. Will have 2 method less but I don't know how it will affect readability. Thing to consider

Reply:
I did consider this approach as well. However, I didn't follow that way since:

- The client code invoking the `GetErrorMap()` actually method decides about the "implementation" for each `Fault` type.
- As mentioned by you, the code readability suffers